galsim/scene.py

```diff
@@ -86,46 +92,58 @@ class COSMOSCatalog(object):
-    @param dir          The directory of catalog file. [default: None]
-    @param preload      Keyword that is only used for real galaxies, not parametric ones, to choose
-                        whether to preload the header information.  If `preload=True`, the bulk of  
-                        the I/O time is in the constructor.  If `preload=False`, there is
-                        approximately the same total I/O time (assuming you eventually use most of
-                        the image files referenced in the catalog), but it is spread over the
-                        various calls to getGal() and getPSF().  [default: False]
-    @param noise_dir    Keyword that is only used for real galaxies, not parametric ones.
-                        The directory of the noise files if different from the directory of the 
-                        image files.  [default: image_dir]
-    @param use_real     Enable the use of realistic galaxies?  [default: True]
-                        If this parameter is False, then `makeGalaxy(gal_type='real')` will
-                        not be allowed.
-    @param exclude_fail Exclude galaxies that have failures in the parametric fits? [default: True]
-    @param exclude_bad  Exclude those that have evidence of probably being a bad fit?  e.g. n > 5
-                        and hlr > 1 arcsec, probably indicates poor sky subtraction. [default: True]
-    @param min_hlr      Exclude galaxies whose fitted half-light-radius is smaller than this value
-                        (in arcsec).  [default: 0, meaning no limit]
-    @param max_hlr      Exclude galaxies whose fitted half-light-radius is larger than this value
-                        (in arcsec).  [default: 0, meaning no limit]
-    @param min_flux     Exclude galaxies whose fitted flux is smaller than this value.  
-                        [default: 0, meaning no limit]
-    @param max_flux     Exclude galaxies whose fitted flux is larger than this value.  
-                        [default: 0, meaning no limit]
+    @param file_name        The file containing the catalog. [default: None, which will look for the
+                            F814W<25.2 COSMOS catalog in $PREFIX/share/galsim.  It will raise an
+                            exception if the catalog is not there telling you to run
+                            galsim_download_cosmos.]
+    @param sample           A keyword argument that can be used to specify the sample to use, i.e.,
+                            "23.5" or "25.2".  At most one of `file_name` and `sample` should be
+                            specified.
+                            [default: None, which results in the same default as `file_name=None`.]
+    @param image_dir        Keyword that is only used for real galaxies, not parametric ones, to
+                            specify the directory of the image files.
+                            If a string containing no `/`, it is the relative path from the location
```

Comment:
I've always found the way these dir kwargs are parsed a bit confusing, to be honest.  I'd actually rather have just a single dir for everything and let the user use symlinks if they want to put things in different directories than the standard layout.  Seems to me that would be less confusing than having 3 different dir kwargs here.  Then there wouldn't be any need to distinguish between absolute and relative paths.